src/main/java/seedu/contax/ui/onboarding/OnboardingWindow.java

```diff
@@ -119,126 +117,270 @@ public void translate(double x, double y) {
-        personListPanel = new PersonListPanel(persons.asUnmodifiableObservableList());
-        personListPanelPlaceholder.getChildren().add(personListPanel.getRoot());
-
-        commandBoxPlaceholder.getChildren().add(commandBox.getRoot());
+    private void coverAll() {
+        overlay.cover(parentPane.layoutXProperty(), parentPane.layoutYProperty(),
+                parentPane.heightProperty(), parentPane.widthProperty());
     }
 
     /**
-     * Process the given OnboardingStep and translate it to stage
-     * @param step The OnboardingStep to be processed
+     * Initialize the onboarding story manager by processing the first step and settings up event listeners
+     * @param mng
      */
-    public void processStep(OnboardingStep step) {
-        if (step != null) {
-            instructionLabel.setText(step.getDisplayMessage());
-            instructionLabel.setSize(step.getMessageHeight(),
-                    step.getMessageWidth(), stage.heightProperty(), stage.widthProperty());
-            processOverlayOption(step.getOverlayOption());
-            processHighlightOption(step.getHighlightOption());
-            if (step.getPersonOperation() == 1) {
-                persons.add(step.getPerson());
-            } else if (step.getPersonOperation() == 2) {
-                mainWindow.show();
-                stage.hide();
+    private void initStoryManager(OnboardingStoryManager mng) {
+        processStep(storyManager.start());
+        parentPane.setOnMouseClicked((event -> {
+            OnboardingStep s = storyManager.handleMouseClick();
+            if (s != null) {
+                processStep(s);
+                processInstructionPosition(s.getPositionOption());
             }
+        }));
 
-            if (step.getCommand() != null) {
-                if (!commandBox.getText().equals(step.getCommand())) {
-                    if (!commandBox.getText().equals("")) {
-                        instructionLabel.setText("Please type: " + step.getCommand());
-                    }
-                    return;
-                } else {
-                    OnboardingStep s = storyManager.getNextStep();
+        commandBox.getRoot().setOnKeyPressed((event) -> {
+            if (event.getCode() == KeyCode.ENTER) {
+                OnboardingStep s = storyManager.handleEnter();
+                if (s != null) {
                     processStep(s);
-                    step.setEventType(s.getPositionOption());
+                    processInstructionPosition(s.getPositionOption());
                 }
             }
-            storyManager.stepFront();
-        }
-
+        });
     }
 
     /**
-     * Highlight the stage based on the given option
+     * Highlights a node in this window the stage based on the given option
+     * <br><br>
+     * Options are as follows:
+     * <br>- 0: clear all highlights
+     * <br>- 1: highlight only the command box
+     * <br>- 2: highlight only the person list
```

Comment:
Would be better if you abstract it into an Enum, which can be nested in OnboardingStory if necessary.